src/features/administrative/scripts/import-statistics.ts

```diff
@@ -10,83 +10,80 @@
-  const femalePopulations = []
-  const totalPopulations = []
-  const birthRates = []
-
-  for (const year of sourceStatistic.years) {
-    femalePopulations.push({
-      [year.year]: year.female_population
-    })
-    malePopulations.push({
-      [year.year]: year.male_population
-    })
-    totalPopulations.push({
-      [year.year]: year.population
-    })
-    birthRates.push({
-      [year.year]: year.crude_birth_rate / 2
-    })
-  }
-
-  const extensions: fhir.Extension[] = [
-    {
-      url: 'http://opencrvs.org/specs/id/statistics-male-populations',
-      valueString: JSON.stringify(malePopulations)
-    },
-    {
-      url: 'http://opencrvs.org/specs/id/statistics-female-populations',
-      valueString: JSON.stringify(femalePopulations)
-    },
-    {
-      url: 'http://opencrvs.org/specs/id/statistics-total-populations',
-      valueString: JSON.stringify(totalPopulations)
-    },
-    {
-      url: 'http://opencrvs.org/specs/id/statistics-crude-birth-rates',
-      valueString: JSON.stringify(birthRates)
-    }
-  ]
-  return extensions
-}
+import { readCSVToJSON, sendToFhir } from '@countryconfig/features/utils'
+import { Location } from '@countryconfig/scripts/validate-source-files'
+import { z } from 'zod'
+import {
+  extractLocationTree,
+  extractMaxAdminLevel,
+  extractStatisticsMap,
+  generateStatisticalExtensions,
+  getChildLocations,
+  getLocationByIdentifier,
+  getStatistics,
+  mergeLocationStatistics
+} from './utils'
 
 async function addStatisticalData() {
-  const statistics = await getStatistics(process.argv[2])
-  // tslint:disable-next-line:no-console
+  const stats = await getStatistics(process.argv[2])
+  const rawLocations = (await readCSVToJSON(process.argv[3])) as Array<
+    z.infer<typeof Location>
+  >
+  const maxAdminLevel = extractMaxAdminLevel(rawLocations as any)
+  const locationMap = extractLocationTree(rawLocations as any, maxAdminLevel)
+  const statisticMap = extractStatisticsMap(stats)
+
+  // eslint-disable-next-line no-console
   console.log(
     `${chalk.blueBright(
       '/////////////////////////// UPDATING LOCATIONS WITH STATISTICAL DATA IN FHIR ///////////////////////////'
     )}`
   )
 
-  for (const statistic of statistics) {
+  const alreadyUpdatedLocations: string[] = []
+
+  // adminLevel 0 doesn't need to be added as it's the country level
+  // If it needs to be added in the future, amend the condition to be `adminLevel >= 0`
+  for (let adminLevel = maxAdminLevel; adminLevel > 0; adminLevel--) {
+    for (let i = 0; i < rawLocations.length; i++) {
+      const column = `admin${adminLevel}Pcode`
+      const location = rawLocations[i][column]!
+      const name = rawLocations[i][`admin${adminLevel}Name_alias`]!
+
+      if (alreadyUpdatedLocations.includes(location)) {
+        continue
+      }
+
+      let statisticsForLocation = statisticMap.get(location)
+      if (!statisticsForLocation) {
+        const children = getChildLocations(location, locationMap)
```

Comment:
... then children would be null as the lowest level has no children

Reply:
see other comment